core/frontend/src/Viewport.ts

```diff
@@ -2494,6 +2494,51 @@ export abstract class Viewport implements IDisposable, TileUser {
     return this.target.readImageToCanvas();
   }
 
+  /** Waits for all tiles to load and render for this viewport.
+   * @returns A promise which will resolve when all tiles finish loading and are rendered for this viewport.
+   * @param options The options governing how this method behaves. Particularly, if the `isAddedToViewManager` property is false (defaults to true), this method will pump the tile request scheduler itself.
+   */
+  public async waitForTilesToLoad(options: {isAddedToViewManager: boolean} = {isAddedToViewManager: true}): Promise<void> {
```

Comment:
Remove `isAddedToViewManager` argument - ask ViewManager if it knows about this viewport.